packages/fast-colors/src/colorlib/colorPalette.ts

```diff
@@ -0,0 +1,271 @@
+                : ColorPalette.defaultPaletteConfig.multiplyDark,
+        };
+
+        this.palette = [];
+        this.updatePaletteColors();
+    }
+
+    private readonly config: IColorPaletteConfig;
+    public readonly palette: ColorRGBA64[];
+
+    public updatePaletteGenerationValues(newConfig: IColorPaletteConfig): boolean {
+        let changed: boolean = false;
+        if (
+            newConfig.interpolationMode &&
+            newConfig.interpolationMode !== this.config.interpolationMode
+        ) {
+            this.config.interpolationMode = newConfig.interpolationMode;
+            changed = true;
+        }
+        if (
+            newConfig.scaleColorLight &&
+            !newConfig.scaleColorLight.equalValue(this.config.scaleColorLight)
+        ) {
+            this.config.scaleColorLight = newConfig.scaleColorLight;
+            changed = true;
+        }
+        if (
+            newConfig.scaleColorDark &&
+            !newConfig.scaleColorDark.equalValue(this.config.scaleColorDark)
+        ) {
+            this.config.scaleColorDark = newConfig.scaleColorDark;
+            changed = true;
+        }
+        if (newConfig.clipLight && newConfig.clipLight !== this.config.clipLight) {
+            this.config.clipLight = newConfig.clipLight;
+            changed = true;
+        }
+        if (newConfig.clipDark && newConfig.clipDark !== this.config.clipDark) {
+            this.config.clipDark = newConfig.clipDark;
+            changed = true;
+        }
+        if (
+            newConfig.saturationAdjustmentCutoff &&
+            newConfig.saturationAdjustmentCutoff !==
+                this.config.saturationAdjustmentCutoff
+        ) {
+            this.config.saturationAdjustmentCutoff = newConfig.saturationAdjustmentCutoff;
+            changed = true;
+        }
+        if (
+            newConfig.saturationLight &&
+            newConfig.saturationLight !== this.config.saturationLight
+        ) {
+            this.config.saturationLight = newConfig.saturationLight;
+            changed = true;
+        }
+        if (
+            newConfig.saturationDark &&
+            newConfig.saturationDark !== this.config.saturationDark
+        ) {
+            this.config.saturationDark = newConfig.saturationDark;
+            changed = true;
+        }
+        if (
+            newConfig.overlayLight &&
+            newConfig.overlayLight !== this.config.overlayLight
+        ) {
+            this.config.overlayLight = newConfig.overlayLight;
+            changed = true;
+        }
+        if (newConfig.overlayDark && newConfig.overlayDark !== this.config.overlayDark) {
+            this.config.overlayDark = newConfig.overlayDark;
+            changed = true;
+        }
+        if (
+            newConfig.multiplyLight &&
+            newConfig.multiplyLight !== this.config.multiplyLight
+        ) {
+            this.config.multiplyLight = newConfig.multiplyLight;
+            changed = true;
+        }

```

Comment:
I wonder if these if conditionals could be consolidated into something like `checkColorPaletteConfigValue(ColorPaletteConfigPropertyKey.multiplyLight)` where ColorPaletteConfigPropertyKey is an enum like:
```
enum ColorPaletteConfigPropertyKey {
    multiplyLight = "multiplyLight"
    ...etc.
}
```